pkg/features/README.md

```diff
@@ -6,30 +6,41 @@ Currently, feature flags use environment variables to power the toggles.
 
 Feature flags can be valuable to ship features in a preview state, to provide the end user a way to disable some functionality or to control any boolean setting.
 
-> :warning: Feature flags cannot be used inside migrations or schema changes.
-> Migrations must be merged to the master branch without any feature flag gate, and must not break any current features.
+## Use cases
 
-## Adding a feature flag
+### Feature release stages
 
-To add a feature flag, add a variable with your feature to `list.go`. To register this feature flag variables, you are required to provide the following:
+To promote a feature through various the release stages, consider the following progression:
 
-* Name: This is a short description of what this flag is about
-* Environment Variable name: This is the environment variable which needs to be set to override the default value of this flag. Env var names **must** start with `ROX_`
-* Options:
-  * whether the flag is changeable on release builds (default) or not (add `unchangeableInProd` option). See below for more details
-  * default value: disabled (default) or enabled (add `enabled` option)
+1. Until the feature is ready to be tested by customers, the flag should be disabled by default and unchangeable in production.
+2. The flag, disabled by default, can be made changeable to allow customers share feedback on the feature.
+3. When the feature is complete and generally available, the flag can be enabled by default.
+4. After the feature has been tested in production for some time, but not longer, remove the flag.
+
+### Feature deprecation
+
+When an existing feature needs to be removed, a feature flag should be used to gradually deprecate the feature:
 
-The variable can be one of two types of feature flag:
+1. Create a feature flag, disabled by default, with a variable having `HIDDEN`, `SUPPRESSION` or similar word in the name. Do not use `DISABLED`, because the condition for the enabled feature will read bad.
```

Comment:
```suggestion
1. Create a feature flag, disabled by default, with a variable having `HIDDEN`, `SUPPRESSION` or similar word in the name. Do not use `DISABLED`, because the condition for the enabled feature will read badly.
```